### PR TITLE
Implement `--failure-mode` option

### LINKED ISSUE #90: `-failure-mode` option?
Would you consider a PR to pass a `-failure-mode` option to `llvm-profdata merge`?
https://llvm.org/docs/CommandGuide/llvm-profdata.html#cmdoption-llvm-profdata-merge-failure-mode

### PR DESCRIPTION
**Pull request by smoelius**
Fixes #90.
I modified the existing merge test. I hope that was okay.
I am open to changing anything in this PR, no matter how small.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -29,6 +29,7 @@ fs-err = "2.5"
 glob = "0.3"
 home = "0.5"
 is_executable = "1"
+itertools = "0.10.1"
```
Comment (taiki-e) [inline Cargo.toml:32]: This seems to be used only in tests. Could you move this to dev-dependencies?

**Review by taiki-e — COMMENTED** [review]
Thanks!

File: Cargo.toml (hunk 1 above)
Comment (smoelius) [reply Cargo.toml]: Sure. (Sorry, silly mistake.)

**Review by taiki-e — APPROVED** [review]
Thanks!

bors r+

**Comment by taiki-e** [comment]
Published in v0.1.7.

**Pull request merged**